src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -102,7 +102,23 @@ def _is_validation_done(self) -> bool:
     @property
     def done(self) -> bool:
         """Evaluates when to leave the loop."""
-        return (self._is_training_done and self._is_validation_done) or self.trainer.should_stop
+        if self._is_training_done and self._is_validation_done:
+            return True
+
+        if self.trainer.should_stop:
+            # early stopping
+            min_epochs = self.trainer.fit_loop.min_epochs
+            met_min_epochs = self.trainer.current_epoch >= min_epochs if min_epochs else True
```

Comment:
I would create protected propertiees for these checks, so we reuse the code in both loops

Reply:
done!